data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/RequestMapper.java

```diff
@@ -180,10 +184,16 @@ public void handle(final HttpServerRequest request) {
     producerProps.setProperty(CloudEventSerializer.EVENT_FORMAT_CONFIG, JsonFormat.CONTENT_TYPE);
 
     // Get the rc and increment it
-    final ReferenceCounter<ProducerHolder<K, V>> rc = this.producerReferences.computeIfAbsent(
-      producerProps,
-      props -> new ReferenceCounter<>(new ProducerHolder<>(producerCreator.apply(props)))
-    );
+    if (!this.producerReferences.containsKey(producerProps)) {
+      try {
+        final var producer = producerCreator.apply(producerProps);
+        this.producerReferences.put(producerProps, new ReferenceCounter<>(new ProducerHolder<>(producer)));
+      } catch (final Exception ex) {
+        return Future.failedFuture(ex);
+      }
+    }
+
+    final ReferenceCounter<ProducerHolder<K, V>> rc = this.producerReferences.get(producerProps);
```

Comment:
Why removing the computeIfAbsent? If there is an exception in `producerCreator.apply(props)` it should be propagated outside too https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/util/Map.html#computeIfAbsent(K,java.util.function.Function):

> If the mapping function returns null, no mapping is recorded. If the mapping function itself throws an (unchecked) exception, the exception is rethrown, and no mapping is recorded.

Reply:
Ok, thanks!

Reply:
Done!